openhtf/plugs/frontline/bluetooth_sniffer.py

```diff
@@ -0,0 +1,277 @@
+
+  @contextlib.contextmanager
+  def sniff(self, bluetooth_mac, capture_id):
+    """Start and stop the sniffer through the context manager interface."""
+    self.start_sniffing(bluetooth_mac, capture_id)
+
+    try:
+      yield
+    finally:
+      self.stop_sniffing()
+
+
+def _init_logging(logger, level):
+  """Logging setup for use when this module is invoked as runnable."""
+  logger.setLevel(logging.DEBUG)
+  log_to_stderr = logging.StreamHandler()
+  log_to_stderr.setLevel(level)
+  log_to_stderr.setFormatter(
+      logging.Formatter('%(asctime)s.%(msecs)03d %(levelname)s: %(message)s',
+                        '%Y-%m-%d %H:%M:%S'))
+  logger.addHandler(log_to_stderr)
+
+
+def _parse_args():
+  """Argument parsing for use when this module is invoked as runnable."""
+  parser = argparse.ArgumentParser(
+      description=__doc__ + '\nScript version: ' + VERSION,
+      formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument('--version', action='version', version=VERSION)
+  parser.add_argument(
+      '-v', '--verbose', action='store_true', help='enable verbose logging')
+  parser.add_argument(
+      '-a',
+      '--address',
+      required=True,
+      help='Bluetooth MAC address of the bluetooth device or phone to sniff.')
+  parser.add_argument(
+      '--host',
+      default='localhost',
+      help="The sniffer's automation server address.")
+  parser.add_argument('--capture_id', default='sniffer', help='capture ID')
+  parser.add_argument(
+      '-p',
+      '--port',
+      type=int,
+      default=DEFAULT_AUTOMATION_PORT,
+      help="The sniffer's automation server port.")
+  parser.add_argument(
+      '-l',
+      '--local_log_path',
+      type=str,
+      default='',
+      help='The path to sniffer logs on the local machine, if applicable.')
+  parser.add_argument(
+      '-d',
+      '--download_port',
+      type=int,
+      default=DEFAULT_HTTP_PORT,
+      help='The port for the HTTP server with the sniffer log files.')
+  parser.add_argument(
+      '-o',
+      '--output',
+      default='ble_sniffer_data',
+      help='Filename at which to output the collected sniffer data.')
+  parser.add_argument(
+      '-s',
+      '--seconds',
+      type=float,
+      help='Time to run; If not specified, runs until stopped.')
+  return parser.parse_args()
+
+
+def _run_bluetooth_sniffer(host, port, local_log_path, download_port, address,
+                           verbose, capture_id, output, seconds):
+  """Run the bluetooth sniffer; for use when module is invoked as runnable.
+
+  Args: All arguments correspond to the matching args defined directly above in
+      the _parse_args function. See help strings there for details.
+  """
+  _init_logging(_LOG, logging.DEBUG if verbose else logging.INFO)
+
+  if not os.path.exists(output):
+    os.makedirs(output)
+
+  sniffer = BleSniffer(host, port)
+  try:
+    sniffer.start_sniffing(address, capture_id)
+    if seconds is None:
```

Comment:
since 0 for seconds is a useless value, might be more readable to do `if seconds:` and invert the blocks.  up to you though if you'd prefer to keep the stricter check.

Reply:
Done.